File: src/sorting/BubbleSort.java
package sorting;

public class BubbleSort {

	void bubbleSort(int arr[]) {
		int size = arr.length;
		for (int pass = size - 1; pass >= 0; pass--) { //in every pass largest element moves to the last
			for (int j = 0; j < pass; j++) {
				if (arr[j] > arr[j + 1]) {
					//swap elements
					int temp = arr[j];
					arr[j] = arr[j + 1];
					arr[j + 1] = temp;
				}
			}
		}
	}
}

File: src/main/kotlin/algos/arrays/sorting/BubbleSort.kt
package algos.arrays.sorting

class BubbleSort {

    fun sort(array: Array<Int>): Array<Int> {
        if(array.isEmpty()) return emptyArray()

        var end = array.lastIndex

        while (end >= 0) {

            (0..end).forEach{ index ->
                if (index == array.lastIndex) return@forEach

                if (array[index] > array[index + 1]) {
                    swap(array, index, index+1)
                }
            }

            end--
        }

        return array;
    }


    private fun swap(arr: Array<Int>, i: Int, j: Int) {
        val temp = arr[i]
        arr[i] = arr[j]
        arr[j] = temp
    }
}

fun main(args: Array<String>) {
    val arr = arrayOf(2, 1, 3)
    println(arr.contentToString())

    BubbleSort().sort(arr)

    println("\nsorted:")
    println(arr.contentToString())
}